### PR TITLE
Resolve State Label Rather than Status Label

### PR DESCRIPTION
**Pull request by sskulkarni**
### Description

We have displayed State Label rather than Status Label on Order status Grid and in a drop down in Assign Status to state.
For that I have converted State Code to State Label with some PHP function,As we don't have State Label input in Magento2.
Find screen-shots for reference.
orderstatus_grid
orderstate_dropdown

Please consider "0204934" commit for this solution.

### Fixed Issues (if relevant)

1. magento/magento2#17847: Wrong State Title, Displaying Status Label Rather than State
2. magento/magento2#17844: Order State dropdown in Assign Order Status to State incorrect

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17850) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @sskulkarni. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by sskulkarni** [comment]
I am changing my email address and make pull request again.
Thanks

**Comment by sskulkarni** [comment]
I have changed Email Address.
Thanks

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function __construct(
     public function execute()
     {
         $itemId = (int)$this->getRequest()->getParam('item_id');
-        $itemQty = (int)$this->getRequest()->getParam('item_qty');
+        $itemQty = (float)$this->getRequest()->getParam('item_qty');
```
Comment (osrecio) [inline app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: This change is related with issues or PR?

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 1 above)
Comment (sskulkarni) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: No
Thanks

Comment (osrecio) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: Can you revert it? To focus only in the issue. Thanks!

**Comment by osrecio** [comment]
Hi @sskulkarni . You delete the entire file: UpdateItemQty.php. You only should revert the change of this commit: https://github.com/magento/magento2/pull/17850/files/020493427d1bd76fb86dc9c4be376c7871fcdd87#diff-1533c00e78677683edc94872df589b5f.

When you solve it. Please squash your commits to only have one.

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 1 above)
Comment (sskulkarni) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: I have reverted.
Thanks

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function __construct(
     public function execute()
     {
         $itemId = (int)$this->getRequest()->getParam('item_id');
-        $itemQty = (int)$this->getRequest()->getParam('item_qty');
```
Comment (osrecio) [inline app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: This file should not be modified. Please leave it as it was originally. I mentioned it in my previous comment.

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 2 above)
Comment (sskulkarni) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: Done

**Review by sskulkarni — COMMENTED** [review]
Reverted file

File: app/code/Magento/Sales/Model/Order/Config.php [hunk 3]
```diff
@@ -181,15 +181,27 @@ public function getStatuses()
...
+                $states[$item->getState()] = __($this->getStateLabelByCode($item->getState()));
             }
         }
         return $states;
     }
 
+    /**
+     * Convert State Code to State Label
+     */
+    public function getStateLabelByCode($stateCode)
```
Comment (slavvka) [inline app/code/Magento/Sales/Model/Order/Config.php:198]: You cannot add public methods to classes and interfaces marked with @api. Please reimplement using the guide https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/

**Comment by slavvka** [comment]
Hey @sskulkarni are you planning to complete this PR?

File: app/code/Magento/Sales/Model/Order/Config.php (hunk 3 above)
Comment (sivaschenko) [reply app/code/Magento/Sales/Model/Order/Config.php:198]: @slavvka please consider that this is not an interface, adding public methods to class shouldn't result in any negative consequences.

**Comment by sivaschenko** [comment]
Closed due to inactivity. Feel free to reopen if you'd like to continue progress on this PR.

**Pull request closed without merging**